drivers/sensor/bq274xx/bq274xx.c

```diff
@@ -387,265 +606,42 @@ static int gauge_init(const struct device *dev)
-	if (ret < 0) {
-		LOG_ERR("Failed to write designEnergy MSB");
-		return -EIO;
-	}
-
-	ret = i2c_reg_write_byte_dt(&config->i2c, BQ274XX_EXT_BLKDAT_DESIGN_ENR_LOW,
-				    designenergy_lsb);
-	if (ret < 0) {
-		LOG_ERR("Failed to write designEnergy LSB");
-		return -EIO;
-	}
-
-	ret = i2c_reg_write_byte_dt(&config->i2c, BQ274XX_EXT_BLKDAT_TERMINATE_VOLT_HIGH,
-				    terminatevolt_msb);
-	if (ret < 0) {
-		LOG_ERR("Failed to write terminateVolt MSB");
-		return -EIO;
-	}
-
-	ret = i2c_reg_write_byte_dt(&config->i2c, BQ274XX_EXT_BLKDAT_TERMINATE_VOLT_LOW,
-				    terminatevolt_lsb);
-	if (ret < 0) {
-		LOG_ERR("Failed to write terminateVolt LSB");
-		return -EIO;
-	}
-
-	ret = i2c_reg_write_byte_dt(&config->i2c, BQ274XX_EXT_BLKDAT_TAPERRATE_HIGH,
-				    taperrate_msb);
-	if (ret < 0) {
-		LOG_ERR("Failed to write taperRate MSB");
-		return -EIO;
-	}
-
-	ret = i2c_reg_write_byte_dt(&config->i2c, BQ274XX_EXT_BLKDAT_TAPERRATE_LOW, taperrate_lsb);
-	if (ret < 0) {
-		LOG_ERR("Failed to write taperRate LSB");
-		return -EIO;
-	}
-
-	for (uint8_t i = 0; i < 32; i++) {
-		block[i] = 0;
-	}
-
-	ret = read_data_block(dev, 0x00, block, 32);
-	if (ret < 0) {
-		LOG_ERR("Unable to read block data");
-		return -EIO;
-	}
-
-	checksum_new = 0;
-	for (uint8_t i = 0; i < 32; i++) {
-		checksum_new += block[i];
-	}
-	checksum_new = 255 - checksum_new;
-
-	ret = cmd_reg_write(dev, BQ274XX_EXT_CHECKSUM, checksum_new);
-	if (ret < 0) {
-		LOG_ERR("Failed to update new checksum");
-		return -EIO;
-	}
-
-	tmp_checksum = 0;
-	ret = i2c_reg_read_byte_dt(&config->i2c, BQ274XX_EXT_CHECKSUM, &tmp_checksum);
-	if (ret < 0) {
-		LOG_ERR("Failed to read checksum");
-		return -EIO;
-	}
-
-	ret = ctrl_reg_write(dev, BQ274XX_CTRL_BAT_INSERT);
-	if (ret < 0) {
-		LOG_ERR("Unable to configure BAT Detect");
-		return -EIO;
-	}
-
-	ret = ctrl_reg_write(dev, BQ274XX_CTRL_SOFT_RESET);
-	if (ret < 0) {
-		LOG_ERR("Failed to soft reset the gauge");
-		return -EIO;
-	}
-
-	flags = 0;
-	/* Poll Flags   */
-	do {
-		ret = cmd_reg_read(dev, BQ274XX_CMD_FLAGS, &flags);
-		if (ret < 0) {
-			LOG_ERR("Unable to read flags");
-			return -EIO;
-		}
-
-		if (flags & 0x0010) {
-			k_msleep(BQ274XX_SUBCLASS_DELAY * 10);
-		}
-	} while (flags & 0x0010);
-
-	/* Seal the gauge */
-	ret = ctrl_reg_write(dev, BQ274XX_CTRL_SEALED);
-	if (ret < 0) {
-		LOG_ERR("Failed to seal the gauge");
-		return -EIO;
-	}
-
-	data->configured = true;
-
-	return 0;
-}
-
 #ifdef CONFIG_BQ274XX_PM
-static int enter_shutdown_mode(const struct device *dev)
+static int bq274xx_enter_shutdown_mode(const struct device *dev)
 {
 	int ret;
 
-	ret = ctrl_reg_write(dev, BQ274XX_UNSEAL_KEY);
+	ret = bq274xx_ctrl_reg_write(dev, BQ274XX_UNSEAL_KEY);
 	if (ret < 0) {
 		LOG_ERR("Unable to unseal the battery");
 		return ret;
 	}
 
-	ret = ctrl_reg_write(dev, BQ274XX_UNSEAL_KEY);
+	ret = bq274xx_ctrl_reg_write(dev, BQ274XX_UNSEAL_KEY);
 	if (ret < 0) {
 		LOG_ERR("Unable to unseal the battery");
 		return ret;
 	}
```

Comment:
Same duplication?

Reply:
Nice! I did not notice, so the technical manual says: "The Sealed to Unsealed key has two identical
words", so yeah it's intentional but since it's supposed to be two different keys I changed this to two different defines with the same value so it's somewhat clear-er it's intentional. Looking at the Linux driver (https://lxr.missinglinkelectronics.com/linux/drivers/power/supply/bq27xxx_battery.c#L1006) it seems like some device have to different keys, but I guess this will do for now.